command/base_flags.go

```diff
@@ -30,6 +30,86 @@ type FlagBool interface {
 	IsBoolFlag() bool
 }
 
+// BoolPtr is a bool which is aware if it has been set.
+type BoolPtr struct {
+	v *bool
+}
+
+func (b *BoolPtr) Set(v string) error {
+	if b.v == nil {
+		b.v = new(bool)
+	}
+	var err error
+	*(b.v), err = strconv.ParseBool(v)
+	return err
+}
+
+func (b *BoolPtr) IsSet() bool {
+	return b.v != nil
+}
+
+func (b *BoolPtr) Get() bool {
+	if b.v == nil {
+		return false
+	}
+	return *b.v
+}
+
+func (b *BoolPtr) String() string {
+	var current bool
+	if b.v != nil {
+		current = *(b.v)
+	}
+	return fmt.Sprintf("%v", current)
+}
+
+type boolPtrValue struct {
+	target *BoolPtr
+}
+
+func newBoolPtrValue(def *bool, target *BoolPtr) *boolPtrValue {
+	val := &boolPtrValue{
+		target: target,
+	}
+	if def != nil {
+		val.target.Set(strconv.FormatBool(*def))
+	}
+	return val
+}
+
+func (b *boolPtrValue) IsBoolFlag() bool {
+	return true
+}
+
+func (b *boolPtrValue) Set(s string) error {
+	if b.target == nil {
+		b.target = new(BoolPtr)
+	}
+	return b.target.Set(s)
+}
+
+func (b *boolPtrValue) String() string { return b.target.String() }
+
+type BoolPtrVar struct {
+	Name       string
+	Aliases    []string
+	Usage      string
+	Hidden     bool
```

Comment:
If you're going to define a Hidden field you may as well implement FlagVisibility.

Reply:
Hidden and EnvVar is now supported. 

> Also should there be a new FlagSet method BoolPtrVar?
Yep. Its already there.

Reply:
Haha my early morning reading skills aren't the best it seems.